### PR TITLE
Fix build with GCC 10, change manpage section of bumblebeed, set ENABLE_PRIMUS_LAYER for primus-vk support

### PR DESCRIPTION
**Pull request by bluca**
Patches forwarded from Debian

### CODE REVIEW

**Review by amonakov — APPROVED** [review]

**Review by amonakov — COMMENTED** [review]
Can you ask the author of primus-vk to comment on this? I think we should ask if he is happy with this.

**Review by ArchangeGabriel — APPROVED** [review]

File: src/bbsecondary.c [hunk 1]
```diff
@@ -139,10 +139,10 @@ bool start_secondary(bool need_secondary) {
     return true;
   //no problems, start X if not started yet
   if (!bb_is_running(bb_status.x_pid)) {
-    char pci_id[13];
+    char pci_id[16];
     static char *x_conf_file;
     // 0-255 bus, 0-31 slot, 0-7 func
-    snprintf(pci_id, 13, "PCI:%03d:%02d:%o", pci_bus_id_discrete->bus,
+    snprintf(pci_id, 16, "PCI:%03d:%02d:%o", pci_bus_id_discrete->bus,
```
Comment (amonakov) [inline src/bbsecondary.c]: Would have been nice to use `sizeof pci_id` instead of `16` here (like it was already the case in other places).

**Comment by amonakov** [comment]
I am happy with these, except for the third commit which adds implicit primus-vk enabling: can we ask the opinion of the author of primus-vk on this?

**Comment by bluca** [comment]
> I am happy with these, except for the third commit which adds implicit primus-vk enabling: can we ask the opinion of the author of primus-vk on this?

This change is a result of collaboration with the author, so I think we are ok (eg: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=892646 )

@felixdoerre any objections? We are shipping bumblebee in Debian/Ubuntu with this

**Comment by ArchangeGabriel** [comment]
@felixdoerre ^

Those changes are fine, I think they are some other that could be integrated (https://github.com/archlinux/svntogit-community/blob/packages/bumblebee/trunk/0007-bb_mutebblogger.patch, https://github.com/archlinux/svntogit-community/blob/packages/bumblebee/trunk/0008-libglvnd.patch).

More generally I’m wondering whether we should add a new “transport” like [this](https://github.com/archlinux/svntogit-packages/blob/packages/nvidia-prime/trunk/prime-run). On Arch they are open requests to change Bumblebee that way and completely ditch primus/primus_vk, only keeping Bumblebee for the power management part (since it requires a Turing+ and CoffeeLake+ system to work with proprietary drivers).

**Comment by bluca** [comment]
> @felixdoerre ^
> 
> Those changes are fine, I think they are some other that could be integrated (https://github.com/archlinux/svntogit-community/blob/packages/bumblebee/trunk/0007-bb_mutebblogger.patch, https://github.com/archlinux/svntogit-community/blob/packages/bumblebee/trunk/0008-libglvnd.patch).

Will double-check thanks, I think we have something akin to the former already, and I thought the latter wasn't needed anymore, but not 100% sure.

> More generally I’m wondering whether we should add a new “transport” like [this](https://github.com/archlinux/svntogit-packages/blob/packages/nvidia-prime/trunk/prime-run). On Arch they are open requests to change Bumblebee that way and completely ditch primus/primus_vk, only keeping Bumblebee for the power management part (since it requires a Turing+ and CoffeeLake+ system to work with proprietary drivers).

It would be really great and I gave that some thought in the past, but I _think_ glx/xorg side is not ready for it? As far as I can understand, the way offloading is implemented means the driver needs to be permanently loaded, and the xserver cannot drop the "reference". Not even wayland+xwayland-on-demand worked last time I tried. I think I had spotted some work-in-progress to allow that to happen, but I don't have a source right now. I might be completely wrong though, as I haven't really spent much time on it, just some quick experiments.

File: src/bbsecondary.c (hunk 1 above)
Comment (bluca) [reply src/bbsecondary.c]: Thanks, fixed

**Comment by ArchangeGabriel** [comment]
@bluca Hum, now that you say that I’m having doubts too. I’ll check with the person that started this discussion on Arch and report. Anyway this is not completely related to this PR, so we can still handle this later.

**Review by felixdoerre — APPROVED** [review]
Yes, I am happy with these changes.

**Comment by ArchangeGabriel** [comment]
So I can confirm that for PRIME offloading the driver has to be loaded before X, and that subsequent attempts to unbind it do not work. At this point this means they are still no good solution for PM + performances for pre-Turing or pre-Coffee Lake systems.

**Pull request merged**